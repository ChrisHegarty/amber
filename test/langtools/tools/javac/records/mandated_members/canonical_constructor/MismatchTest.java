/*
 * @test /nodynamiccopyright/
 * @summary check that the compiler doesn't accept canonical constructors with name mismatch
 * @compile/fail/ref=MismatchTest.out -XDrawDiagnostics MismatchTest.java
 */

import java.util.*;

public class MismatchTest {
    record R1(int i, int j) {
        public R1(int j, int i) {} // doesn't match by name
    }

    record R2(int i, List<String> ls) {
        public R2(int i, List ls) {}
    }
}
