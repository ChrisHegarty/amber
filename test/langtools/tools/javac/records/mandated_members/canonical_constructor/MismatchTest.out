MismatchTest.java:11:16: compiler.err.canonical.with.name.mismatch
MismatchTest.java:15:16: compiler.err.constructor.with.same.erasure.as.canonical
- compiler.note.unchecked.filename: MismatchTest.java
- compiler.note.unchecked.recompile
2 errors
